pallets/lbp/src/lib.rs

```diff
@@ -740,6 +768,21 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	/// Validates fee collector
+	/// Not more than one fee collector per asset
+	/// This is done to prevent race conditions with locks on the same asset
+	fn validate_fee_collector(
+		fee_collector: T::AccountId,
```

Comment:
If you still need this helper function after the change to use try_mutate. 

I'd consider either remove it completely because it is just one ensure and inline it.

Or use a reference for the collector param, so no need to copy everywhere.

Reply:
benefit of the function can be testability, where you can check that this behaves correctly with some storage state, without any complex storage setup

Reply:
where you would just set storage and check if the result is expected, without any pool creation etc ...